src/Simplex/FileTransfer/Transport.hs

```diff
@@ -40,18 +46,33 @@ sendFile h send = go
   where
     go 0 = pure ()
     go sz =
-      B.hGet h xftpBlockSize >>= \case
-        "" -> ioe_EOF
-        ch -> do
-          let ch' = B.take (fromIntegral sz) ch -- sz >= xftpBlockSize
-          send $ byteString ch'
-          go $ sz - fromIntegral (B.length ch')
+      getFileChunk h sz >>= \ch -> do
+        send $ byteString ch
+        go $ sz - fromIntegral (B.length ch)
+
+sendEncFile :: Handle -> (Builder -> IO ()) -> LC.SbState -> Word32 -> IO ()
+sendEncFile h send = go
+  where
+    go sbState 0 = do
+      -- TODO remove padding when HTTP2 issue is fixed
+      let authTag = BA.convert (LC.sbAuth sbState) <> B.replicate (xftpBlockSize - C.authTagSize) '#'
+      send $ byteString authTag
+    go sbState sz =
+      getFileChunk h sz >>= \ch -> do
+        let (encCh, sbState') = LC.sbEncryptChunk sbState ch
+        send (byteString encCh) `E.catch` \(e :: E.SomeException) -> print e >> E.throwIO e
+        -- TODO remove delay when HTTP2 issue is fixed
+        threadDelay 500
```

Comment:
what's the HTTP2 issue that requires padding and delay?

Reply:
https://github.com/kazu-yamamoto/http2/issues/40